include/deal.II/matrix_free/matrix_free.templates.h

```diff
@@ -1788,6 +1788,254 @@ MatrixFree<dim, Number, VectorizedArrayType>::initialize_indices(
                   }
               }
           }
+
+
+          // partitioner 3: values on faces
```

Comment:
What comes next is quite a lot of code. Do you see a chance to combine it with the other code somehow, e.g. by a separate function or by a lambda? I have not looked into it in detail yet, but 90% of code seems very similar.

Reply:
I will try to introduce a lambda.